### PR TITLE
Make clear where reducers act on a state slice instead of on whole state

### PR DESCRIPTION
**Pull request by hjvanranden**
Make clear where reducers act on a state slice instead of on the whole state, by renaming their parameter 'state' to 'stateSlice'.

Motivation: My colleague and I were independently confused on the content of parameter 'state' in the reducers that act on a state slice.

### CODE REVIEW

**Comment by timdorr** [comment]
Thanks!

**Comment by markerikson** [comment]
I'm not entirely sure I agree with the naming change here.

Part of the idea of reducers like this is that as far as the function is considered, this _is_ the entire state.  The fact that it's just one slice is a concern of the parent reducer function, not this one.  In fact, you could reuse the function in another context, or even as a root reducer.

File: docs/basics/Reducers.md [hunk 1]
```diff
@@ -396,27 +396,27 @@ import {
 } from './actions'
 const { SHOW_ALL } = VisibilityFilters
 
-function visibilityFilter(state = SHOW_ALL, action) {
+function visibilityFilter(stateSlice = SHOW_ALL, action) {
   switch (action.type) {
     case SET_VISIBILITY_FILTER:
-      return action.filter
+      return stateSlice.filter
```
Comment (naw) [inline docs/basics/Reducers.md]: This change here is incorrect. cc @timdorr

**Comment by naw** [comment]
Perhaps call it `visibilityState` in the visibility reducer and `todosState` in the todos reducer? This might address both of your concerns, @markerikson and @hjvanranden

**Pull request merged**